core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -975,6 +975,9 @@ private void doScheduleAsyncCleanupRequest(Backoff cleanupBackoff, Request reque
             public void onSuccess(JsonResponse<TaskInfo> result)
             {
                 try {
+                    if (!result.hasValue()) {
+                        throw new IllegalArgumentException("TaskInfo result did not contain JSON payload; payload=" + result.getResponseBody());
```

Comment:
or should `result.getResponseBody()` be logged instead?
also, you may want to hex encode

Reply:
Yeah - logging is probaly safer - will change. I would rather not do hex as it is more painful to use.

Reply:
not very hard. and the problem may be about some non-printable stuff (or invalid unicode) and therefore harder to see in plain logs

Reply:
you can also do both: plain  + hex